subsys/fs/fs.c

```diff
@@ -148,6 +148,10 @@ int fs_open(struct fs_file_t *zfp, const char *file_name, fs_mode_t flags)
 	}
 
 	zfp->mp = mp;
+	if (((mp->flags & FS_MOUNT_FLAG_READ_ONLY) == FS_MOUNT_FLAG_READ_ONLY) &&
```

Comment:
I really think this should be `(mp->flags & FS_MOUNT_FLAG_READ_ONLY_MASK) != 0)` or something else that only tests the read-only bit.  This part of the code doesn't care whether the file system was flagged with no-format.  That the two flags are combined are only a convenience for the user, one that I'm not convinced is justified.

It's reasonable IMO to keep the flags completely separate, and simply error if somebody tries to mount a file system read-only in a situation where it would have to be initialized.  Here having the wrapper implicitly add the no-format flag when read-only seems less confusing.